components/function-controller/pkg/apis/serverless/v1alpha1/function_types.go

```diff
@@ -27,6 +36,10 @@ type FunctionSpec struct {
 
 	// +optional
 	Labels map[string]string `json:"labels,omitempty"`
+
+	SourceType SourceType `json:"type"`
```

Comment:
Shouldn't this field be optional for the backward compatibility?

Reply:
yeah, please make it a pointer and add kubebuilder markers